urllib3/_collections.py

```diff
@@ -94,11 +138,12 @@ def clear(self):
                 self.dispose_func(value)
 
     def keys(self):
+        # type: () -> List[_K]
```

Comment:
Should return a valueview, a set like interface, so it is not really a replacement for a dict, maybe an issue here? We could also have it return a set handling duplicated values.